states/statemgr/helper.go

```diff
@@ -5,8 +5,37 @@ package statemgr
 
 import (
 	"github.com/hashicorp/terraform/states"
+	"github.com/hashicorp/terraform/states/statefile"
+	"github.com/hashicorp/terraform/version"
 )
 
+// NewStateFile creates a new statefile.File object, with a newly-minted
+// lineage identifier and serial 0, and returns a pointer to it.
+func NewStateFile() *statefile.File {
+	return &statefile.File{
+		Lineage:          NewLineage(),
+		TerraformVersion: version.SemVer,
+	}
+}
+
+// StateFile is a special helper to obtain a statefile representation
+// of a state snapshot that can be written later by a call
+func StateFile(mgr Storage, state *states.State) *statefile.File {
+	ret := &statefile.File{
+		State: state.DeepCopy(),
+	}
```

Comment:
It seems like `TerraformVersion` doesn't get populated in this codepath. I guess that doesn't matter right now because we just use this to create a throwaway object for testing against, as we discussed, but maybe better to assign that in here just in case we forget and use this function for some other purpose in the future.

Reply:
Let me quickly add that...

Reply:
Done